server/app/services/applicant/ApplicantService.java

```diff
@@ -107,7 +109,8 @@ public ApplicantService(
       HttpExecutionContext httpExecutionContext,
       DeploymentType deploymentType,
       ServiceAreaUpdateResolver serviceAreaUpdateResolver,
-      EsriClient esriClient) {
+      EsriClient esriClient,
```

Comment:
I did some looking into this and I think this can be achieved by adding a new Esri module: https://www.playframework.com/documentation/2.8.x/JavaDependencyInjection#Programmatic-bindings

For example, the file storage provider strategy implementation is provided based on a value in the config: https://github.com/civiform/civiform/blob/main/server/app/modules/CloudStorageModule.java.  We could provider the FakeEsriClient if DeploymentType.isDevOrStaging() is true

Reply:
Building on what Avritt said, rather than inject both EsriClient and FakeEsriClient I think we should do one of the following:
* Create a [custom Guice module](https://www.playframework.com/documentation/2.8.x/JavaDependencyInjection#Programmatic-bindings) to allow the dependency injection framework to inject the appropriate type (recommended). This will allow just having to inject the one interface.
* Create a factory that can determine the appropriate object to use, and inject the factory.

Reply:
Thanks for the great suggestion @avrittrohwer and @gwendolyngoetz! Added an EsriModule and EsriClient is now an abstract class. Went with a class instead of an interface for EsriClient, since `getAddressSuggestions` and `getServiceAreaInclusionGroup` methods are the same between RealEsriClient and FakeEsriClient.